platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/GuestClaimsHandlerExt.java

```diff
@@ -13,129 +13,208 @@
 
-    private Map<String, Object> claimsProfiles;
+    private Map<String, String> availableClaimsMap;
 
-    private List<String> availableClaims;
+    private List<Map<String, String>> claimsProfileMaps;
 
-    //holds names of claims that if modified would cause guest login failures
-    private List<String> immutableClaims;
+    private List<Map<String, String>> claimsProfileBannerMaps;
 
-    private String profileDir;
+    private final PropertiesFileSupport propertiesFileSupport;
 
-    private String availableClaimsFile;
+    // ** Holds names of claims that if modified would cause guest login failures **
+    private final List<String> immutableClaims;
 
-    public GuestClaimsHandlerExt() {
-        immutableClaims = new ArrayList<>();
-    }
+    private final String profilesDirectory;
+
+    private final String profilesBannerDirectory;
+
+    private final String availableClaimsFile;
+
+    private String selectedClaimsProfileName;
 
     /**
-     * Load the profiles from disk
+     * Constructor that enforces immutability where possible.
      */
-    private Map<String, Object> loadClaimsProfiles(String profileDirPath) {
-        Map<String, Object> claimsProfilesMap = new HashMap<>();
-        if (!StringUtils.isEmpty(profileDirPath)) {
-            File dir = new File(profileDirPath);
-            if (dir.exists()) {
-                File[] propertyFiles = dir.listFiles((dir1, name) -> {
-                    return name.endsWith(".properties");
-                });
-                if (propertyFiles != null) {
-                    for (File profileFile : propertyFiles) {
-                        Map<String, String> map = loadPropertiesFile(profileFile);
-                        String profileName = map.get(PROFILE_NAME_KEY);
-                        if (!StringUtils.isEmpty(profileName)) {
-                            map.remove(PROFILE_NAME_KEY);
-                            claimsProfilesMap.put(profileName, convertMapToList(map));
-                        }
-                    }
-                }
-            }
-        }
-        return claimsProfilesMap;
-    }
+    public GuestClaimsHandlerExt(PropertiesFileSupport propertiesFileSupport,
+            List<String> immutableClaims, String profilesDirectory, String profilesBannerDirectory,
+            String availableClaimsFile) {
+
+        this.propertiesFileSupport = propertiesFileSupport;
+        this.immutableClaims = immutableClaims;
+        this.profilesDirectory = profilesDirectory;
+        this.profilesBannerDirectory = profilesBannerDirectory;
+        this.availableClaimsFile = availableClaimsFile;
 
-    private List<String> convertMapToList(Map<String, String> map) {
-        List<String> list = new ArrayList<>();
-        for (Map.Entry<String, String> entry : map.entrySet()) {
-            list.add(entry.getKey() + "=" + entry.getValue());
-        }
-        return list;
+        selectedClaimsProfileName = null;
```

Comment:
Yes but I want to make it explicit that this is to be expected. The state _should_ start out as `null` and be replaced later.